backend/src/core/services/upload.service.ts

```diff
@@ -11,19 +13,33 @@ import { MissingTemplateKeysError } from '@core/errors/template.errors'
 import { configureEndpoint } from '@core/utils/aws-endpoint'
 import { jwtUtils } from '@core/utils/jwt'
 import { Campaign } from '@core/models'
-import { CsvStatusInterface } from '@core/interfaces'
+import {
+  CsvStatusInterface,
+  UploadData,
+  Upload,
+  AllowedTemplateTypes,
+} from '@core/interfaces'
 import { CSVParams } from '@core/types'
-import { StatsService, CampaignService } from '.'
+import S3Client from '@core/services/s3-client.class'
+import { StatsService, CampaignService, ParseCsvService } from '.'
 
 const logger = loggerWithLabel(module)
 const MAX_PROCESSING_TIME = config.get('csvProcessingTimeout')
```

Comment:
do we still need this? is the job guaranteed to fail if the process promise is rejected?

Reply:
I didn't set an explicit timeout on the `bee-queue` job, so this means that if the promise is rejected, it is from the job failing. However, I think we should probably still keep this to prevent jobs from being stuck in the processing state. 

Didn't try to fix the timeout in this PR because it would need quite a bit of change. In order to handle it properly, we will need to be able to abort/interrupt the CSV processing mid-parse once the timeout is reached.